rgw/admin/quota.go

```diff
@@ -0,0 +1,56 @@
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+)
+
+// Quota describes an object store quota for a user or a bucket
+type Quota struct {
+	UID        string `json:"user_id" url:"uid"`
+	QuotaType  string `url:"quota-type"`
+	Enabled    *bool  `json:"enabled" url:"quota,enabled"`
+	CheckOnRaw bool   `json:"check_on_raw"`
+	MaxSize    *int64 `json:"max_size" url:"max-size"`
+	MaxSizeKb  *int64 `json:"max_size_kb" url:"max-size-kb"`
+	MaxObjects *int64 `json:"max_objects" url:"max-objects"`
+}
```

Comment:
What about quota on `Maxbuckets` for a user? As far I understand this is supported,  see the `bucket` option in [Quota section](https://docs.ceph.com/en/octopus/radosgw/adminops/#quotas)

The quota on the bucket count is needed from OBC provisioner in Rook

Reply:
Max bucket will be set when updating the user using ModifyUser()